lib/bricks-net-http-impl.js

```diff
@@ -0,0 +1,858 @@
+
+var ConnectionResetByPeer = require('./bricks-net-exceptions').ConnectionResetByPeer;
+var AttemptedToSendHTTPResponseMoreThanOnce = require('./bricks-net-exceptions').AttemptedToSendHTTPResponseMoreThanOnce;
+
+var DefaultInternalServerErrorMessage = require('./bricks-net-http-default_messages').DefaultInternalServerErrorMessage;
+
+var HTTPResponseCode = require('./bricks-net-http-codes').HTTPResponseCode;
+var HTTPResponseCodeAsString = require('./bricks-net-http-codes').HTTPResponseCodeAsString;
+
+var HTTPHeaders = require('./bricks-net-http-headers').HTTPHeaders;
+
+
+var kCRLF = "\r\n";
+var kCRLFLength = kCRLF.length;
+var kHeaderKeyValueSeparator = ": ";
+var kHeaderKeyValueSeparatorLength = kHeaderKeyValueSeparator.length;
+var kContentLengthHeaderKey = "Content-Length";
+var kTransferEncodingHeaderKey = "Transfer-Encoding";
+var kTransferEncodingChunkedValue = "chunked";
+
+
+function TemplatedHTTPRequestData(connection) {
+	var _this = this;
+	
+	promiseHelpers.makeThenable(_this);
+	
+	Object.defineProperties(_this, {
+		_DEBUG_LOG: {
+			configurable: true,
+			enumerable: false,
+			value:
+				/* istanbul ignore next: debug method */
+				function () {}
+				//function () { console.log.apply(console, arguments); }
+		},
+		toString: {
+			configurable: true,
+			enumerable: false,
+			value: function () {
+				return 'TemplatedHTTPRequestData';
+			}
+		},
+		
+		// private:
+		method_: {
+			enumerable: false,
+			writable: true,
+			value: ""
+		},
+		url_: {
+			enumerable: false,
+			writable: true,
+			value: ""
+		},
+		raw_path_: {
+			enumerable: false,
+			writable: true,
+			value: ""
+		},
+		buffer_: {
+			enumerable: false,
+			writable: true,
+			value: null
+		},
+		body_buffer_begin_: {
+			enumerable: false,
+			writable: true,
+			value: null
+		},
+		body_buffer_end_: {
+			enumerable: false,
+			writable: true,
+			value: null
+		},
+		prepared_body_: {
+			enumerable: false,
+			writable: true,
+			value: null
+		},
+		chunked_body_buffer_: {
+			enumerable: false,
+			writable: true,
+			value: null
+		}
+	});
+	
+	// Read the response right in the constructor (like in the original code, but async).
+	(function () {
+		_this._DEBUG_LOG(_this + ': Begin.');
+		
+		function strstr(buffer, str, start) {
+			//_this._DEBUG_LOG(_this + ' strstr:',
+			//	inspect(buffer.toString('utf8')),
+			//	inspect(str),
+			//	start
+			//);
+			
+			var i, ic, j, jc;
+			for (ic = buffer.length, i = start || 0; i < ic; ++i) {
+				if (buffer[i] === 0) {
+					break;
+				}
+				for (jc = str.length, j = 0; j < jc; ++j) {
+					//_this._DEBUG_LOG(_this + ' strstr: ',
+					//	i + j, buffer[i + j], j, str.charCodeAt(j), buffer[i + j] === str.charCodeAt(j));
+					if (buffer[i + j] !== str.charCodeAt(j)) {
+						break;
+					}
+				}
+				if (j === jc) {
+					//_this._DEBUG_LOG(_this + ' strstr: return ', i);
+					return i;
+				}
+			}
+			return -1;
+		}
+		
+		
+		var UINT_MAX = 4294967295;
```

Comment:
Fixed in https://github.com/sompylasar/KnowSheet-REPL/commit/2cd4130dfe41529ab867929de228b5a535dc089b